internal/checks/checks.go

```diff
@@ -204,16 +209,29 @@ func (c *Updater) Run(ctx context.Context) error {
 				Msg("context cancelled, closing updater")
 			return nil
 
+		case errors.Is(err, errProbeUnregistered):
+			// Probe unregistered itself from the API, wait
+			// until attempting to reconnect again.
+			c.logger.Warn().
+				Str("connection_state", c.api.conn.GetState().String()).
+				Msg("unregistered probe in API, sleeping for 1 minute...")
```

Comment:
thinking about reconnect here.

`errProbeUnregistered` means that we disconnected when `SIGUSR1` fired. i.e. we told probe to disconnect but keep running checks.

but if we reconnect back in a minute, `SIGUSR1` will `disconnect probe for a minute` and not `disconnect` forever.

I think one minute is reasonable time to reconnect back but maybe we should make it configurable.

Reply:
If we find that the timeout needs to be configured we can make it configurable once we get there.

1 minute should be enough for the other agent to connect and start running, but I know there have been instances where new agents take longer to do that. We need to experiment and figure out what's the appropriate strategy.